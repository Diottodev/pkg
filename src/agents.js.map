{"version":3,"file":"agents.js","sourceRoot":"","sources":["agents.ts"],"names":[],"mappings":"AAAA,SAAS,MAAM,CAAC,KAAa;IAC3B,OAAO,CAAC,IAAc,EAAE,EAAE;QACxB,IAAI,IAAI,CAAC,MAAM,GAAG,CAAC;YACjB,OAAO,GAAG,KAAK,QAAQ,IAAI,CAAC,CAAC,CAAC,OAAO,IAAI,CAAC,KAAK,CAAC,CAAC,CAAC,CAAC,IAAI,CAAC,GAAG,CAAC,EAAE,CAAA;;YAC3D,OAAO,GAAG,KAAK,QAAQ,IAAI,CAAC,CAAC,CAAC,EAAE,CAAA;IACvC,CAAC,CAAA;AACH,CAAC;AAED,MAAM,IAAI,GAAG;IACX,OAAO,EAAE,UAAU;IACnB,KAAK,EAAE,cAAc;IACrB,SAAS,EAAE,kBAAkB;IAC7B,QAAQ,EAAE,gCAAgC;IAC1C,QAAQ,EAAE,qBAAqB;IAC/B,KAAK,EAAE,cAAc;IACrB,SAAS,EAAE,kBAAkB;IAC7B,qBAAqB,EAAE,8BAA8B;IACrD,SAAS,EAAE,SAAS;IACpB,WAAW,EAAE,iBAAiB;IAC9B,kBAAkB,EAAE,wBAAwB;CAC7C,CAAA;AAED,MAAM,IAAI,GAAG;IACX,OAAO,EAAE,UAAU;IACnB,KAAK,EAAE,cAAc;IACrB,SAAS,EAAE,YAAY;IACvB,QAAQ,EAAE,0BAA0B;IACpC,QAAQ,EAAE,iBAAiB;IAC3B,KAAK,EAAE,cAAc;IACrB,SAAS,EAAE,iBAAiB;IAC5B,qBAAqB,EAAE,oBAAoB;IAC3C,SAAS,EAAE,cAAc;IACzB,WAAW,EAAE,iBAAiB;IAC9B,kBAAkB,EAAE,0BAA0B;CAC/C,CAAA;AAED,MAAM,GAAG,GAAG;IACV,OAAO,EAAE,SAAS;IAClB,KAAK,EAAE,aAAa;IACpB,SAAS,EAAE,iBAAiB;IAC5B,QAAQ,EAAE,uBAAuB;IACjC,QAAQ,EAAE,gBAAgB;IAC1B,KAAK,EAAE,aAAa;IACpB,SAAS,EAAE,gBAAgB;IAC3B,qBAAqB,EAAE,gBAAgB;IACvC,SAAS,EAAE,UAAU;IACrB,WAAW,EAAE,gBAAgB;IAC7B,kBAAkB,EAAE,mBAAmB;CACxC,CAAA;AAED,MAAM,CAAC,MAAM,MAAM,GAAG;IACpB,KAAK,EAAE;QACL,OAAO,EAAE,SAAS;QAClB,KAAK,EAAE,MAAM,CAAC,KAAK,CAAC;QACpB,SAAS,EAAE,WAAW;QACtB,QAAQ,EAAE,QAAQ;QAClB,QAAQ,EAAE,cAAc;QACxB,KAAK,EAAE,WAAW;QAClB,SAAS,EAAE,gBAAgB;QAC3B,qBAAqB,EAAE,IAAI;QAC3B,SAAS,EAAE,SAAS;QACpB,WAAW,EAAE,mBAAmB;QAChC,kBAAkB,EAAE,sBAAsB;KAC3C;IACD,MAAM,EAAE,IAAI;IACZ,YAAY,EAAE;QACZ,GAAG,IAAI;QACP,QAAQ,EAAE,0BAA0B;QACpC,SAAS,EAAE,aAAa;QACxB,qBAAqB,EAAE,gBAAgB;QACvC,SAAS,EAAE,cAAc;QACzB,4EAA4E;QAC5E,QAAQ,EAAE,cAAc;QACxB,kBAAkB,EAAE,sBAAsB;KAC3C;IACD,MAAM,EAAE,IAAI;IACZ,qBAAqB;IACrB,QAAQ,EAAE;QACR,GAAG,IAAI;QACP,GAAG,EAAE,MAAM,CAAC,MAAM,CAAC;KACpB;IACD,KAAK,EAAE,GAAG;CACX,CAAA;AAKD,MAAM,CAAC,MAAM,MAAM,GAAG,MAAM,CAAC,IAAI,CAAC,MAAM,CAAY,CAAA;AAEpD,wDAAwD;AACxD,MAAM,CAAC,MAAM,KAAK,GAA0B;IAC1C,WAAW,EAAE,KAAK;IAClB,gBAAgB,EAAE,MAAM;IACxB,WAAW,EAAE,MAAM;IACnB,mBAAmB,EAAE,KAAK;IAC1B,qBAAqB,EAAE,KAAK;CAC7B,CAAA;AAED,MAAM,CAAC,MAAM,YAAY,GAA0B;IACjD,KAAK,EAAE,gBAAgB;IACvB,MAAM,EAAE,8BAA8B;IACtC,QAAQ,EAAE,kCAAkC;IAC5C,MAAM,EAAE,6CAA6C;IACrD,YAAY,EAAE,6CAA6C;IAC3D,KAAK,EAAE,uDAAuD;CAC/D,CAAA"}