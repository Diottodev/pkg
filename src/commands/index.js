export * from '../index';
//# sourceMappingURL=index.js.map