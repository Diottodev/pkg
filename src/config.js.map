{"version":3,"file":"config.js","sourceRoot":"","sources":["config.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,MAAM,SAAS,CAAA;AACxB,OAAO,IAAI,MAAM,WAAW,CAAA;AAC5B,OAAO,OAAO,MAAM,cAAc,CAAA;AAClC,OAAO,GAAG,MAAM,KAAK,CAAA;AAErB,OAAO,EAAE,MAAM,EAAE,MAAM,UAAU,CAAA;AAEjC,MAAM,YAAY,GAAG,OAAO,CAAC,GAAG,CAAC,cAAc,CAAA;AAE/C,MAAM,IAAI,GAAG,OAAO,CAAC,QAAQ,KAAK,OAAO;IACvC,CAAC,CAAC,OAAO,CAAC,GAAG,CAAC,WAAW;IACzB,CAAC,CAAC,OAAO,CAAC,GAAG,CAAC,IAAI,CAAA;AAEpB,MAAM,aAAa,GAAG,IAAI,CAAC,IAAI,CAAC,IAAI,IAAI,IAAI,EAAE,OAAO,CAAC,CAAA;AAEtD,MAAM,MAAM,GAAG,YAAY,IAAI,aAAa,CAAA;AAO5C,MAAM,aAAa,GAAW;IAC5B,YAAY,EAAE,QAAQ;IACtB,WAAW,EAAE,KAAK;CACnB,CAAA;AAED,IAAI,MAA0B,CAAA;AAE9B,MAAM,CAAC,KAAK,UAAU,SAAS;IAC7B,IAAI,CAAC,MAAM,EAAE;QACX,MAAM,KAAK,GAAG,MAAM,MAAM,CAAC,EAAE,YAAY,EAAE,IAAI,EAAE,CAAC,CAAA;QAClD,IAAI,KAAK;YACP,MAAM,GAAG,EAAE,GAAG,aAAa,EAAE,YAAY,EAAE,KAAK,EAAE,CAAA;;YAElD,MAAM,GAAG,MAAM,CAAC,MAAM,CAAC,EAAE,EAAE,aAAa,EAAE,GAAG,CAAC,KAAK,CAAC,EAAE,CAAC,YAAY,CAAC,MAAM,EAAE,OAAO,CAAC,CAAC,CAAC,CAAA;KACzF;IAED,OAAO,MAAgB,CAAA;AACzB,CAAC;AAED,MAAM,CAAC,KAAK,UAAU,eAAe,CAAC,YAAsB;IAC1D,MAAM,EAAE,YAAY,EAAE,GAAG,MAAM,SAAS,EAAE,CAAA;IAC1C,IAAI,YAAY,KAAK,QAAQ,IAAI,CAAC,YAAY,IAAI,OAAO,CAAC,GAAG,CAAC,EAAE,CAAC;QAC/D,OAAO,KAAK,CAAA;IACd,OAAO,YAAY,CAAA;AACrB,CAAC;AAED,MAAM,CAAC,KAAK,UAAU,cAAc;IAClC,MAAM,EAAE,WAAW,EAAE,GAAG,MAAM,SAAS,EAAE,CAAA;IACzC,OAAO,WAAW,CAAA;AACpB,CAAC"}