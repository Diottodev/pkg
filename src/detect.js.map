{"version":3,"file":"detect.js","sourceRoot":"","sources":["detect.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,MAAM,SAAS,CAAA;AACxB,OAAO,IAAI,MAAM,WAAW,CAAA;AAC5B,OAAO,OAAO,MAAM,cAAc,CAAA;AAClC,OAAO,EAAE,YAAY,EAAE,MAAM,OAAO,CAAA;AACpC,OAAO,EAAE,MAAM,EAAE,MAAM,SAAS,CAAA;AAChC,OAAO,YAAY,MAAM,eAAe,CAAA;AACxC,OAAO,OAAO,MAAM,gBAAgB,CAAA;AAEpC,OAAO,EAAE,MAAM,EAAE,YAAY,EAAE,KAAK,EAAE,MAAM,UAAU,CAAA;AACtD,OAAO,EAAE,SAAS,EAAE,MAAM,SAAS,CAAA;AAQnC,MAAM,CAAC,KAAK,UAAU,MAAM,CAAC,EAAE,WAAW,EAAE,YAAY,EAAE,GAAG,KAAoB,EAAE;IACjF,IAAI,KAAK,GAAiB,IAAI,CAAA;IAC9B,IAAI,OAAO,GAAkB,IAAI,CAAA;IAEjC,MAAM,QAAQ,GAAG,MAAM,MAAM,CAAC,MAAM,CAAC,IAAI,CAAC,KAAK,CAAC,EAAE,EAAE,GAAG,EAAE,CAAC,CAAA;IAC1D,IAAI,eAAmC,CAAA;IAEvC,IAAI,QAAQ;QACV,eAAe,GAAG,IAAI,CAAC,OAAO,CAAC,QAAQ,EAAE,iBAAiB,CAAC,CAAA;;QAE3D,eAAe,GAAG,MAAM,MAAM,CAAC,cAAc,EAAE,EAAE,GAAG,EAAE,CAAC,CAAA;IAEzD,8CAA8C;IAC9C,IAAI,eAAe,IAAI,EAAE,CAAC,UAAU,CAAC,eAAe,CAAC,EAAE;QACrD,IAAI;YACF,MAAM,GAAG,GAAG,IAAI,CAAC,KAAK,CAAC,EAAE,CAAC,YAAY,CAAC,eAAe,EAAE,MAAM,CAAC,CAAC,CAAA;YAChE,IAAI,OAAO,GAAG,CAAC,cAAc,KAAK,QAAQ,EAAE;gBAC1C,MAAM,CAAC,IAAI,EAAE,GAAG,CAAC,GAAG,GAAG,CAAC,cAAc,CAAC,OAAO,CAAC,KAAK,EAAE,EAAE,CAAC,CAAC,KAAK,CAAC,GAAG,CAAC,CAAA;gBACpE,OAAO,GAAG,GAAG,CAAA;gBACb,IAAI,IAAI,KAAK,MAAM,IAAI,MAAM,CAAC,QAAQ,CAAC,GAAG,CAAC,GAAG,CAAC;oBAC7C,KAAK,GAAG,YAAY,CAAA;qBACjB,IAAI,IAAI,KAAK,MAAM,IAAI,MAAM,CAAC,QAAQ,CAAC,GAAG,CAAC,GAAG,CAAC;oBAClD,KAAK,GAAG,QAAQ,CAAA;qBACb,IAAI,IAAI,IAAI,MAAM;oBACrB,KAAK,GAAG,IAAI,CAAA;qBACT,IAAI,CAAC,YAAY;oBACpB,OAAO,CAAC,IAAI,CAAC,+BAA+B,EAAE,GAAG,CAAC,cAAc,CAAC,CAAA;aACpE;SACF;QACD,WAAM,GAAG;KACV;IAED,uBAAuB;IACvB,IAAI,CAAC,KAAK,IAAI,QAAQ;QACpB,KAAK,GAAG,KAAK,CAAC,IAAI,CAAC,QAAQ,CAAC,QAAQ,CAAC,CAAC,CAAA;IAExC,eAAe;IACf,IAAI,KAAK,IAAI,CAAC,SAAS,CAAC,KAAK,CAAC,KAAK,CAAC,GAAG,CAAC,CAAC,CAAC,CAAC,CAAC,IAAI,CAAC,YAAY,EAAE;QAC7D,IAAI,CAAC,WAAW,EAAE;YAChB,OAAO,CAAC,IAAI,CAAC,kBAAkB,KAAK,yCAAyC,CAAC,CAAA;YAE9E,IAAI,OAAO,CAAC,GAAG,CAAC,EAAE;gBAChB,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,CAAA;YAEjB,MAAM,IAAI,GAAG,YAAY,CAAC,KAAK,EAAE,YAAY,CAAC,KAAK,CAAC,CAAC,CAAA;YACrD,MAAM,EAAE,UAAU,EAAE,GAAG,MAAM,OAAO,CAAC;gBACnC,IAAI,EAAE,YAAY;gBAClB,IAAI,EAAE,SAAS;gBACf,OAAO,EAAE,sCAAsC,IAAI,GAAG;aACvD,CAAC,CAAA;YACF,IAAI,CAAC,UAAU;gBACb,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,CAAA;SAClB;QAED,MAAM,YAAY,CAAC,YAAY,KAAK,GAAG,OAAO,CAAC,CAAC,CAAC,IAAI,OAAO,EAAE,CAAC,CAAC,CAAC,EAAE,EAAE,EAAE,EAAE,KAAK,EAAE,SAAS,EAAE,GAAG,EAAE,CAAC,CAAA;KAClG;IAED,OAAO,KAAK,CAAA;AACd,CAAC"}