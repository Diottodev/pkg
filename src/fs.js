import { resolve } from 'node:path';
import fs from 'node:fs';
import process from 'node:process';
export function getPackageJSON(ctx) {
    var _a;
    const cwd = (_a = ctx === null || ctx === void 0 ? void 0 : ctx.cwd) !== null && _a !== void 0 ? _a : process.cwd();
    const path = resolve(cwd, 'package.json');
    if (fs.existsSync(path)) {
        try {
            const raw = fs.readFileSync(path, 'utf-8');
            const data = JSON.parse(raw);
            return data;
        }
        catch (e) {
            if (!(ctx === null || ctx === void 0 ? void 0 : ctx.programmatic)) {
                console.warn('Failed to parse package.json');
                process.exit(1);
            }
            throw e;
        }
    }
}
//# sourceMappingURL=fs.js.map