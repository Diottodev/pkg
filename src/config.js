import fs from 'node:fs';
import path from 'node:path';
import process from 'node:process';
import ini from 'ini';
import { detect } from './detect';
const customRcPath = process.env.NI_CONFIG_FILE;
const home = process.platform === 'win32'
    ? process.env.USERPROFILE
    : process.env.HOME;
const defaultRcPath = path.join(home || '~/', '.nirc');
const rcPath = customRcPath || defaultRcPath;
const defaultConfig = {
    defaultAgent: 'prompt',
    globalAgent: 'npm',
};
let config;
export async function getConfig() {
    if (!config) {
        const agent = await detect({ programmatic: true });
        if (agent)
            config = { ...defaultConfig, defaultAgent: agent };
        else
            config = Object.assign({}, defaultConfig, ini.parse(fs.readFileSync(rcPath, 'utf-8')));
    }
    return config;
}
export async function getDefaultAgent(programmatic) {
    const { defaultAgent } = await getConfig();
    if (defaultAgent === 'prompt' && (programmatic || process.env.CI))
        return 'npm';
    return defaultAgent;
}
export async function getGlobalAgent() {
    const { globalAgent } = await getConfig();
    return globalAgent;
}
//# sourceMappingURL=config.js.map