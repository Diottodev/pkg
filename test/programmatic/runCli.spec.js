import path from 'node:path';
import { tmpdir } from 'node:os';
import fs from 'fs-extra';
import { afterAll, beforeAll, describe, expect, it, vi } from 'vitest';
import { AGENTS, parsePkAg, parsePkg, parsePkx, parsePkUn, parsePkUp, runCli } from '../../src';
let basicLog, errorLog, warnLog, infoLog;
function runCliTest(fixtureName, agent, runner, args) {
    return async () => {
        const cwd = await fs.mkdtemp(path.join(tmpdir(), 'ni-'));
        const fixture = path.join(__dirname, '..', 'fixtures', fixtureName, agent);
        await fs.copy(fixture, cwd);
        await runCli(async (agent, _, ctx) => {
            // we override the args to be test specific
            return runner(agent, args, ctx);
        }, {
            programmatic: true,
            cwd,
            args,
        }).catch((e) => {
            // it will always throw if execa is mocked
            if (e.command)
                expect(e.command).toMatchSnapshot();
            else
                expect(e.message).toMatchSnapshot();
        });
    };
}
beforeAll(() => {
    basicLog = vi.spyOn(console, 'log');
    warnLog = vi.spyOn(console, 'warn');
    errorLog = vi.spyOn(console, 'error');
    infoLog = vi.spyOn(console, 'info');
    vi.mock('execa', async (importOriginal) => {
        const mod = await importOriginal();
        return {
            ...mod,
            execaCommand: (cmd) => {
                // break execution flow for easier snapshotting
                // eslint-disable-next-line no-throw-literal
                throw { command: cmd };
            },
        };
    });
});
afterAll(() => {
    vi.resetAllMocks();
});
const agents = [...Object.keys(AGENTS), 'unknown'];
const fixtures = ['lockfile', 'packager'];
// matrix testing of: fixtures x agents x commands
fixtures.forEach(fixture => describe(fixture, () => agents.forEach(agent => describe(agent, () => {
    /** pk ag */
    it('pk ag', runCliTest(fixture, agent, parsePkAg, []));
    it('pk ag run foo', runCliTest(fixture, agent, parsePkAg, ['run', 'foo']));
    /** pkg */
    it('pkg', runCliTest(fixture, agent, parsePkg, []));
    it('pkg foo', runCliTest(fixture, agent, parsePkg, ['foo']));
    it('pkg foo -D', runCliTest(fixture, agent, parsePkg, ['foo', '-D']));
    it('pkg --frozen', runCliTest(fixture, agent, parsePkg, ['--frozen']));
    it('pkg -g foo', runCliTest(fixture, agent, parsePkg, ['-g', 'foo']));
    /** pkx */
    it('pkx', runCliTest(fixture, agent, parsePkx, ['foo']));
    /** pk up */
    it('pk up', runCliTest(fixture, agent, parsePkUp, []));
    it('pk up -i', runCliTest(fixture, agent, parsePkUp, ['-i']));
    /** pk un */
    it('pk un foo', runCliTest(fixture, agent, parsePkUn, ['foo']));
    it('pk un -g foo', runCliTest(fixture, agent, parsePkUn, ['-g', 'foo']));
    it('no logs', () => {
        expect(basicLog).not.toHaveBeenCalled();
        expect(warnLog).not.toHaveBeenCalled();
        expect(errorLog).not.toHaveBeenCalled();
        expect(infoLog).not.toHaveBeenCalled();
    });
}))));
//# sourceMappingURL=runCli.spec.js.map