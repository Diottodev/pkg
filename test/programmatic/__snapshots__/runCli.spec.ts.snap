// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lockfile > bun > pk ag 1`] = `"bun"`;

exports[`lockfile > bun > pk ag run foo 1`] = `"bun run foo"`;

exports[`lockfile > bun > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > bun > pk un foo 1`] = `"bun remove foo"`;

exports[`lockfile > bun > pk up -i 1`] = `"bun update"`;

exports[`lockfile > bun > pk up 1`] = `"bun update"`;

exports[`lockfile > bun > pkg --frozen 1`] = `"bun install --no-save"`;

exports[`lockfile > bun > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > bun > pkg 1`] = `"bun install"`;

exports[`lockfile > bun > pkg foo -D 1`] = `"bun add foo -d"`;

exports[`lockfile > bun > pkg foo 1`] = `"bun add foo"`;

exports[`lockfile > bun > pkx 1`] = `"bunx foo"`;

exports[`lockfile > npm > pk ag 1`] = `"npm"`;

exports[`lockfile > npm > pk ag run foo 1`] = `"npm run foo"`;

exports[`lockfile > npm > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > npm > pk un foo 1`] = `"npm uninstall foo"`;

exports[`lockfile > npm > pk up -i 1`] = `"Command \\"upgrade-interactive\\" is not support by agent \\"npm\\""`;

exports[`lockfile > npm > pk up 1`] = `"npm update"`;

exports[`lockfile > npm > pkg --frozen 1`] = `"npm ci"`;

exports[`lockfile > npm > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > npm > pkg 1`] = `"npm i"`;

exports[`lockfile > npm > pkg foo -D 1`] = `"npm i foo -D"`;

exports[`lockfile > npm > pkg foo 1`] = `"npm i foo"`;

exports[`lockfile > npm > pkx 1`] = `"npx foo"`;

exports[`lockfile > pnpm > pk ag 1`] = `"pnpm"`;

exports[`lockfile > pnpm > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`lockfile > pnpm > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > pnpm > pk un foo 1`] = `"pnpm remove foo"`;

exports[`lockfile > pnpm > pk up -i 1`] = `"pnpm update -i"`;

exports[`lockfile > pnpm > pk up 1`] = `"pnpm update"`;

exports[`lockfile > pnpm > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`lockfile > pnpm > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > pnpm > pkg 1`] = `"pnpm i"`;

exports[`lockfile > pnpm > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`lockfile > pnpm > pkg foo 1`] = `"pnpm add foo"`;

exports[`lockfile > pnpm > pkx 1`] = `"pnpm dlx foo"`;

exports[`lockfile > pnpm@6 > pk ag 1`] = `"pnpm"`;

exports[`lockfile > pnpm@6 > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`lockfile > pnpm@6 > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > pnpm@6 > pk un foo 1`] = `"pnpm remove foo"`;

exports[`lockfile > pnpm@6 > pk up -i 1`] = `"pnpm update -i"`;

exports[`lockfile > pnpm@6 > pk up 1`] = `"pnpm update"`;

exports[`lockfile > pnpm@6 > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`lockfile > pnpm@6 > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > pnpm@6 > pkg 1`] = `"pnpm i"`;

exports[`lockfile > pnpm@6 > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`lockfile > pnpm@6 > pkg foo 1`] = `"pnpm add foo"`;

exports[`lockfile > pnpm@6 > pkx 1`] = `"pnpm dlx foo"`;

exports[`lockfile > unknown > pk ag 1`] = `"pnpm"`;

exports[`lockfile > unknown > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`lockfile > unknown > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > unknown > pk un foo 1`] = `"pnpm remove foo"`;

exports[`lockfile > unknown > pk up -i 1`] = `"pnpm update -i"`;

exports[`lockfile > unknown > pk up 1`] = `"pnpm update"`;

exports[`lockfile > unknown > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`lockfile > unknown > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > unknown > pkg 1`] = `"pnpm i"`;

exports[`lockfile > unknown > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`lockfile > unknown > pkg foo 1`] = `"pnpm add foo"`;

exports[`lockfile > unknown > pkx 1`] = `"pnpm dlx foo"`;

exports[`lockfile > yarn > pk ag 1`] = `"yarn"`;

exports[`lockfile > yarn > pk ag run foo 1`] = `"yarn run foo"`;

exports[`lockfile > yarn > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > yarn > pk un foo 1`] = `"yarn remove foo"`;

exports[`lockfile > yarn > pk up -i 1`] = `"yarn upgrade-interactive"`;

exports[`lockfile > yarn > pk up 1`] = `"yarn upgrade"`;

exports[`lockfile > yarn > pkg --frozen 1`] = `"yarn install --frozen-lockfile"`;

exports[`lockfile > yarn > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > yarn > pkg 1`] = `"yarn install"`;

exports[`lockfile > yarn > pkg foo -D 1`] = `"yarn add foo -D"`;

exports[`lockfile > yarn > pkg foo 1`] = `"yarn add foo"`;

exports[`lockfile > yarn > pkx 1`] = `"npx foo"`;

exports[`lockfile > yarn@berry > pk ag 1`] = `"yarn"`;

exports[`lockfile > yarn@berry > pk ag run foo 1`] = `"yarn run foo"`;

exports[`lockfile > yarn@berry > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > yarn@berry > pk un foo 1`] = `"yarn remove foo"`;

exports[`lockfile > yarn@berry > pk up -i 1`] = `"yarn upgrade-interactive"`;

exports[`lockfile > yarn@berry > pk up 1`] = `"yarn upgrade"`;

exports[`lockfile > yarn@berry > pkg --frozen 1`] = `"yarn install --frozen-lockfile"`;

exports[`lockfile > yarn@berry > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > yarn@berry > pkg 1`] = `"yarn install"`;

exports[`lockfile > yarn@berry > pkg foo -D 1`] = `"yarn add foo -D"`;

exports[`lockfile > yarn@berry > pkg foo 1`] = `"yarn add foo"`;

exports[`lockfile > yarn@berry > pkx 1`] = `"npx foo"`;

exports[`packager > bun > pk ag 1`] = `"pnpm"`;

exports[`packager > bun > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > bun > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > bun > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > bun > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > bun > pk up 1`] = `"pnpm update"`;

exports[`packager > bun > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > bun > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > bun > pkg 1`] = `"pnpm i"`;

exports[`packager > bun > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > bun > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > bun > pkx 1`] = `"pnpm dlx foo"`;

exports[`packager > npm > pk ag 1`] = `"pnpm"`;

exports[`packager > npm > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > npm > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > npm > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > npm > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > npm > pk up 1`] = `"pnpm update"`;

exports[`packager > npm > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > npm > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > npm > pkg 1`] = `"pnpm i"`;

exports[`packager > npm > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > npm > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > npm > pkx 1`] = `"pnpm dlx foo"`;

exports[`packager > pnpm > pk ag 1`] = `"pnpm"`;

exports[`packager > pnpm > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > pnpm > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > pnpm > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > pnpm > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > pnpm > pk up 1`] = `"pnpm update"`;

exports[`packager > pnpm > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > pnpm > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > pnpm > pkg 1`] = `"pnpm i"`;

exports[`packager > pnpm > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > pnpm > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > pnpm > pkx 1`] = `"pnpm dlx foo"`;

exports[`packager > pnpm@6 > pk ag 1`] = `"pnpm"`;

exports[`packager > pnpm@6 > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > pnpm@6 > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > pnpm@6 > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > pnpm@6 > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > pnpm@6 > pk up 1`] = `"pnpm update"`;

exports[`packager > pnpm@6 > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > pnpm@6 > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > pnpm@6 > pkg 1`] = `"pnpm i"`;

exports[`packager > pnpm@6 > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > pnpm@6 > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > pnpm@6 > pkx 1`] = `"pnpm dlx foo"`;

exports[`packager > unknown > pk ag 1`] = `"pnpm"`;

exports[`packager > unknown > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > unknown > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > unknown > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > unknown > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > unknown > pk up 1`] = `"pnpm update"`;

exports[`packager > unknown > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > unknown > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > unknown > pkg 1`] = `"pnpm i"`;

exports[`packager > unknown > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > unknown > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > unknown > pkx 1`] = `"pnpm dlx foo"`;

exports[`packager > yarn > pk ag 1`] = `"pnpm"`;

exports[`packager > yarn > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > yarn > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > yarn > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > yarn > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > yarn > pk up 1`] = `"pnpm update"`;

exports[`packager > yarn > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > yarn > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > yarn > pkg 1`] = `"pnpm i"`;

exports[`packager > yarn > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > yarn > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > yarn > pkx 1`] = `"pnpm dlx foo"`;

exports[`packager > yarn@berry > pk ag 1`] = `"pnpm"`;

exports[`packager > yarn@berry > pk ag run foo 1`] = `"pnpm run foo"`;

exports[`packager > yarn@berry > pk un -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > yarn@berry > pk un foo 1`] = `"pnpm remove foo"`;

exports[`packager > yarn@berry > pk up -i 1`] = `"pnpm update -i"`;

exports[`packager > yarn@berry > pk up 1`] = `"pnpm update"`;

exports[`packager > yarn@berry > pkg --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > yarn@berry > pkg -g foo 1`] = `"npm i -g foo"`;

exports[`packager > yarn@berry > pkg 1`] = `"pnpm i"`;

exports[`packager > yarn@berry > pkg foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > yarn@berry > pkg foo 1`] = `"pnpm add foo"`;

exports[`packager > yarn@berry > pkx 1`] = `"pnpm dlx foo"`;
